### ISSUE TITLE
[Bug] api container memory leak after upgrading to 0.14.0

### ISSUE THREAD

**Issue by shaxiaozz** [langgenius/dify#11719]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.14.0

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

***Upgrade Dify***
```
cd dify/docker
docker compose down
git pull origin main
docker compose pull
docker compose up -d
```

### ✔️ Expected Behavior

The version was successfully upgraded to 0.14.0, and the web UI interface can be opened normally

### ❌ Actual Behavior

After executing ```docker compose up -d```, the web UI was slow to load. After checking docker stats, I found that the memory usage of the ```docker-api-1``` container had been gradually increasing until all the OS memory was consumed. The host entered a stuck state and was forced to restart. After the restart, this step was repeated, and the memory usage continued to increase. . . .

I observed the logs of ```docker-api-1``` and found no obvious errors, but there were many INFO logs. As follows:
```
" 200 86313
Preparing database migration...
Database migration skipped
[2024-12-17 02:12:19 +0000] [1] [INFO] Starting gunicorn 23.0.0
[2024-12-17 02:12:19 +0000] [1] [INFO] Listening at: http://0.0.0.0:5001 (1)
[2024-12-17 02:12:19 +0000] [1] [INFO] Using worker: gevent
[2024-12-17 02:12:19 +0000] [37] [INFO] Booting worker with pid: 37
[2024-12-17 02:12:19 +0000] [38] [INFO] Booting worker with pid: 38
[2024-12-17 02:12:19 +0000] [39] [INFO] Booting worker with pid: 39
[2024-12-17 02:12:19 +0000] [40] [INFO] Booting worker with pid: 40
[2024-12-17 02:12:19 +0000] [41] [INFO] Booting worker with pid: 41
[2024-12-17 02:12:19 +0000] [42] [INFO] Booting worker with pid: 42
[2024-12-17 02:12:20 +0000] [43] [INFO] Booting worker with pid: 43
[2024-12-17 02:12:20 +0000] [44] [INFO] Booting worker with pid: 44
[2024-12-17 02:12:20 +0000] [45] [INFO] Booting worker with pid: 45
[2024-12-17 02:12:20 +0000] [46] [INFO] Booting worker with pid: 46
[2024-12-17 02:12:20 +0000] [47] [INFO] Booting worker with pid: 47
[2024-12-17 02:12:20 +0000] [48] [INFO] Booting worker with pid: 48
[2024-12-17 02:12:20 +0000] [49] [INFO] Booting worker with pid: 49
[2024-12-17 02:12:20 +0000] [50] [INFO] Booting worker with pid: 50
[2024-12-17 02:12:20 +0000] [51] [INFO] Booting worker with pid: 51
[2024-12-17 02:12:20 +0000] [52] [INFO] Booting worker with pid: 52
[2024-12-17 02:12:20 +0000] [53] [INFO] Booting worker with pid: 53
[2024-12-17 02:12:20 +0000] [54] [INFO] Booting worker with pid: 54
[2024-12-17 02:12:20 +0000] [55] [INFO] Booting worker with pid: 55
[2024-12-17 02:12:20 +0000] [56] [INFO] Booting worker with pid: 56
[2024-12-17 02:12:20 +0000] [57] [INFO] Booting worker with pid: 57
[2024-12-17 02:12:20 +0000] [58] [INFO] Booting worker with pid: 58
[2024-12-17 02:12:20 +0000] [60] [INFO] Booting worker with pid: 60
[2024-12-17 02:12:20 +0000] [59] [INFO] Booting worker with pid: 59
[2024-12-17 02:12:20 +0000] [61] [INFO] Booting worker with pid: 61
[2024-12-17 02:12:20 +0000] [62] [INFO] Booting worker with pid: 62
[2024-12-17 02:12:20 +0000] [63] [INFO] Booting worker with pid: 63
[2024-12-17 02:12:20 +0000] [64] [INFO] Booting worker with pid: 64
[2024-12-17 02:12:21 +0000] [65] [INFO] Booting worker with pid: 65
[2024-12-17 02:12:21 +0000] [66] [INFO] Booting worker with pid: 66
[2024-12-17 02:12:21 +0000] [67] [INFO] Booting worker with pid: 67
[2024-12-17 02:12:21 +0000] [68] [INFO] Booting worker with pid: 68
[2024-12-17 02:12:21 +0000] [69] [INFO] Booting worker with pid: 69
[2024-12-17 02:12:21 +0000] [70] [INFO] Booting worker with pid: 70
[2024-12-17 02:12:21 +0000] [71] [INFO] Booting worker with pid: 71
[2024-12-17 02:12:21 +0000] [72] [INFO] Booting worker with pid: 72
[2024-12-17 02:12:21 +0000] [73] [INFO] Booting worker with pid: 73
[2024-12-17 02:12:21 +0000] [74] [INFO] Booting worker with pid: 74
[2024-12-17 02:12:21 +0000] [75] [INFO] Booting worker with pid: 75
[2024-12-17 02:12:21 +0000] [76] [INFO] Booting worker with pid: 76
[2024-12-17 02:12:21 +0000] [77] [INFO] Booting worker with pid: 77
[2024-12-17 02:12:21 +0000] [78] [INFO] Booting worker with pid: 78
[2024-12-17 02:12:21 +0000] [79] [INFO] Booting worker with pid: 79
[2024-12-17 02:12:21 +0000] [80] [INFO] Booting worker with pid: 80
[2024-12-17 02:12:21 +0000] [81] [INFO] Booting worker with pid: 81
[2024-12-17 02:12:21 +0000] [82] [INFO] Booting worker with pid: 82
[2024-12-17 02:12:21 +0000] [84] [INFO] Booting worker with pid: 84
[2024-12-17 02:12:22 +0000] [83] [INFO] Booting worker with pid: 83
[2024-12-17 02:12:22 +0000] [85] [INFO] Booting worker with pid: 85
[2024-12-17 02:12:22 +0000] [86] [INFO] Booting worker with pid: 86
[2024-12-17 02:12:22 +0000] [87] [INFO] Booting worker with pid: 87
[2024-12-17 02:12:22 +0000] [88] [INFO] Booting worker with pid: 88
[2024-12-17 02:12:22 +0000] [89] [INFO] Booting worker with pid: 89
[2024-12-17 02:12:22 +0000] [90] [INFO] Booting worker with pid: 90
[2024-12-17 02:12:22 +0000] [92] [INFO] Booting worker with pid: 92
[2024-12-17 02:12:22 +0000] [91] [INFO] Booting worker with pid: 91
[2024-12-17 02:12:22 +0000] [93] [INFO] Booting worker with pid: 93
[2024-12-17 02:12:22 +0000] [94] [INFO] Booting worker with pid: 94
[2024-12-17 02:12:22 +0000] [95] [INFO] Booting worker with pid: 95
[2024-12-17 02:12:22 +0000] [96] [INFO] Booting worker with pid: 96
[2024-12-17 02:12:22 +0000] [97] [INFO] Booting worker with pid: 97
[2024-12-17 02:12:22 +0000] [98] [INFO] Booting worker with pid: 98
[2024-12-17 02:12:22 +0000] [99] [INFO] Booting worker with pid: 99
[2024-12-17 02:12:22 +0000] [100] [INFO] Booting worker with pid: 100
[2024-12-17 02:12:22 +0000] [102] [INFO] Booting worker with pid: 102
[2024-12-17 02:12:22 +0000] [103] [INFO] Booting worker with pid: 103
[2024-12-17 02:12:22 +0000] [101] [INFO] Booting worker with pid: 101
[2024-12-17 02:12:22 +0000] [104] [INFO] Booting worker with pid: 104
[2024-12-17 02:12:22 +0000] [105] [INFO] Booting worker with pid: 105
[2024-12-17 02:12:22 +0000] [106] [INFO] Booting worker with pid: 106
[2024-12-17 02:12:22 +0000] [107] [INFO] Booting worker with pid: 107
[2024-12-17 02:12:23 +0000] [108] [INFO] Booting worker with pid: 108
[2024-12-17 02:12:23 +0000] [109] [INFO] Booting worker with pid: 109
[2024-12-17 02:12:23 +0000] [110] [INFO] Booting worker with pid: 110
[2024-12-17 02:12:23 +0000] [111] [INFO] Booting worker with pid: 111
[2024-12-17 02:12:23 +0000] [112] [INFO] Booting worker with pid: 112
[2024-12-17 02:12:23 +0000] [113] [INFO] Booting worker with pid: 113
[2024-12-17 02:12:23 +0000] [114] [INFO] Booting worker with pid: 114
[2024-12-17 02:12:23 +0000] [115] [INFO] Booting worker with pid: 115
[2024-12-17 02:12:23 +0000] [116] [INFO] Booting worker with pid: 116
[2024-12-17 02:12:23 +0000] [118] [INFO] Booting worker with pid: 118
[2024-12-17 02:12:23 +0000] [117] [INFO] Booting worker with pid: 117
[2024-12-17 02:12:23 +0000] [119] [INFO] Booting worker with pid: 119
[2024-12-17 02:12:23 +0000] [120] [INFO] Booting worker with pid: 120
[2024-12-17 02:12:23 +0000] [121] [INFO] Booting worker with pid: 121
[2024-12-17 02:12:23 +0000] [122] [INFO] Booting worker with pid: 122
[2024-12-17 02:12:24 +0000] [124] [INFO] Booting worker with pid: 124
[2024-12-17 02:12:24 +0000] [123] [INFO] Booting worker with pid: 123
[2024-12-17 02:12:24 +0000] [125] [INFO] Booting worker with pid: 125
[2024-12-17 02:12:24 +0000] [126] [INFO] Booting worker with pid: 126
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
```

Currently, I have rolled back to the previous version through ```git checkout 0.13.2``` When I perform the upgrade operation again, this problem will recur stably~

Are there any other ideas or methods for troubleshooting? Can you guide me? Thank you~

**Comment by shaxiaozz** [comment]
@crazywoola Thanks for your reply~
I made another attempt yesterday.
I went to the warehouse to re-pull the latest dify code, put it in another directory on the host, and then restarted dify. I found that it could be started, and I observed that the memory did not increase continuously.

Because we have a habit of tarring the volume data in the docker directory every time we upgrade.

I don't know what happened, I am very confused~
